src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/OppFinder/WhenHandlingGetOppFinderApprovedStandardDetailsRequest.cs

```diff
@@ -117,33 +134,60 @@ public async Task ThenReturnsRegionDetails()
-                result.RegionResults[1].Region.Should().Be("REGION2");
+                result.VersionResults[0].Version.Should().Be("1.0");
+                result.VersionResults[1].Version.Should().Be("1.1");
             }
         }
 
         [Test]
-        public async Task ThenReturnsVersionDetails()
+        [TestCase("eqaProviderContactEmail@email.com", "eqaProviderContactEmail@email.com", "", "OFQUAL", "eqaProviderContactEmail@email.com")]
+        [TestCase("eqaProviderContactName@email.com", "", "eqaProviderContactName@email.com", "OFQUAL", "eqaProviderContactName@email.com")]
+        [TestCase("", "", "", "OFQUAL", "OFQUAL")]
+        [TestCase("eqaProviderContactName", "", "eqaProviderContactName", "OFQUAL", "OFQUAL")]
+        [TestCase("eqaProviderContactEmail", "eqaProviderContactEmail", "", "OFQUAL", "OFQUAL")]
+        public async Task ThenReturnsEqaProvider(string validationRequestEmail, string eqaProviderContactEmail, string eqaProviderContactName, string eqaProviderName, string eqaProvider)
         {
             //Arrange
             var request = new GetOppFinderApprovedStandardDetailsRequest()
             {
                 StandardReference = "ST0001"
             };
 
+            var validationRequest = new ValidationRequest()
+            {
+                Type = "email",
+                Value = validationRequestEmail
+            };
+
+            ValidationHandler validationHandler = new ValidationHandler(new ValidationService());
+
+            _mockOppFinderRepository.Setup(s => s.GetOppFinderApprovedStandardDetails("ST0001")).ReturnsAsync(new OppFinderApprovedStandardDetailsResult
+            {
+                OverviewResult = new OppFinderApprovedStandardOverviewResult()
+                {
+                    StandardCode = 999,
+                    StandardReference = "ST0001",
+                    EqaProviderContactEmail = eqaProviderContactEmail,
+                    EqaProviderContactName = eqaProviderContactName,
+                    EqaProviderName = eqaProviderName
+                },
+            });
+
+          
+            _mockMediator.Setup(s => s.Send(It.Is<ValidationRequest>(x => x.Type == validationRequest.Type && x.Value == validationRequest.Value), It.IsAny<CancellationToken>()))
+                        .ReturnsAsync(await validationHandler.Handle(validationRequest, It.IsAny<CancellationToken>()));
+
             //Act
+
+            _sut = new GetOppFinderApprovedStandardDetailsHandler(_mockLogger.Object, _mockOppFinderRepository.Object, _mockMediator.Object);
             var result = await _sut.Handle(request, new CancellationToken());
 
             //Assert
-            result.VersionResults.Count.Should().Be(2);
-            using (new AssertionScope())
-            {
-                result.VersionResults[0].Version.Should().Be("1.0");
-                result.VersionResults[1].Version.Should().Be("1.1");
-            }
+            result.Should().NotBeNull();
```

Comment:
When doing more than 1 fluid assertion using an AssertionScope is best practice or remove the NotBeNull as the test will fail if it by throwing an exception on the second assert

Reply:
Updated